packages/gitbook/src/components/DocumentView/CodeBlock/CodeBlockRenderer.tsx

```diff
@@ -0,0 +1,117 @@
+import type { DocumentBlockCode } from '@gitbook/api';
+import assertNever from 'assert-never';
+import { useId } from 'react';
+
+import { tcls } from '@/lib/tailwind';
+
+import { CopyCodeButton } from './CopyCodeButton';
+import type { HighlightLine, HighlightToken } from './highlight';
+import { AnnotationPopover } from '../Annotation/AnnotationPopover';
+import type { BlockProps } from '../Block';
+
+import './theme.css';
+import './CodeBlockRenderer.css';
+
+type CodeBlockRendererProps = Pick<BlockProps<DocumentBlockCode>, 'block' | 'style'> & {
+    lines: HighlightLine[];
+};
+
+/**
+ * The logic of rendering a code block from lines.
+ */
+export function ClientCodeBlockRenderer(props: CodeBlockRendererProps) {
```

Comment:
Is this really a "client component" or the common one between server and client?

```suggestion
export function UniversalCodeBlockRenderer(props: CodeBlockRendererProps) {
```

Reply:
Yeah of course it was a typo.